apis/k8ssandra/v1alpha1/cassandraconfig_types.go

```diff
@@ -1041,3 +1049,106 @@ type TrackWarnings struct {
+
+	// +optional
+	Optional *bool `json:"optional,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	Keystore *string `json:"keystore,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	KeystorePassword *string `json:"keystore_password,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	Truststore *string `json:"truststore,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	TruststorePassword *string `json:"truststore_password,omitempty"`
+
+	// +optional
+	Protocol *string `json:"protocol,omitempty"`
+
+	// +optional
+	AcceptedProtocols *[]string `json:"accepted_protocols,omitempty"`
+
+	// +optional
+	Algorithm *string `json:"algorithm,omitempty"`
+
+	// +optional
+	StoreType *string `json:"store_type,omitempty"`
+
+	// +optional
+	CipherSuites *[]string `json:"cipher_suites,omitempty"`
+
+	// default: false
+	// +optional
+	RequireClientAuth *bool `json:"require_client_auth,omitempty"`
+}
+
+type ServerEncryptionOptions struct {
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// +optional
+	Optional *bool `json:"optional,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	Keystore *string `json:"keystore,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	KeystorePassword *string `json:"keystore_password,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	Truststore *string `json:"truststore,omitempty"`
+
+	// Should not be set explicitly in the custom resource.
+	// The operator will generate the right value based on the EncryptionStores field.
+	// +optional
+	TruststorePassword *string `json:"truststore_password,omitempty"`
+
+	// +optional
+	Protocol *string `json:"protocol,omitempty"`
+
+	// +optional
+	AcceptedProtocols *[]string `json:"accepted_protocols,omitempty"`
+
+	// +optional
+	Algorithm *string `json:"algorithm,omitempty"`
+
+	// +optional
+	StoreType *string `json:"store_type,omitempty"`
+
+	// +optional
+	CipherSuites *[]string `json:"cipher_suites,omitempty"`
+
+	// default: false
+	// +optional
+	RequireClientAuth *bool `json:"require_client_auth,omitempty"`
+
+	// default: none
+	// +optional
+	InternodeEncryption *string `json:"internode_encryption,omitempty"`
```

Comment:
This is an enum it seems, from [the docs](https://cassandra.apache.org/doc/latest/cassandra/configuration/cass_yaml_file.html#server_encryption_options):

```
# On outbound connections, determine which type of peers to securely connect to.
        #   The available options are :
        #     none : Do not encrypt outgoing connections
        #     dc   : Encrypt connections to peers in other datacenters but not within datacenters
        #     rack : Encrypt connections to peers in other racks but not within racks
        #     all  : Always use encrypted connections
```

So you could add: 

```
	// +optional
	// +kubebuilder:validation:Enum=none;dc;rack;all
	// +kubebuilder:default=none
```

Reply:
I would take it step further and define a custom type, e.g., `type EncryptionType string` which could be used for both internode and client encryption. Also, no need for a pointer.

Reply:
actually, client to node encryption doesn't have a field that would use this enum, it's only used by server encryption.

Reply:
I've added the enum kubebuilder annotation 👍